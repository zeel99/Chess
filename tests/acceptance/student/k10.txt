--Check queen piece, fail captures
setup_chess(P,1,1)


setup_chess(Q,2,3)

setup_chess(P,3,1)
setup_chess(P,4,2)
setup_chess(P,4,4)

start_game
move_and_capture(2,3,1,1)
move_and_capture(2,3,1,2)
move_and_capture(2,3,1,3)
move_and_capture(2,3,1,4)
move_and_capture(2,3,2,1)
move_and_capture(2,3,2,2)
move_and_capture(2,3,2,3)
move_and_capture(2,3,2,4)
move_and_capture(2,3,3,1)
move_and_capture(2,3,3,2)
move_and_capture(2,3,3,3)
move_and_capture(2,3,3,4)
move_and_capture(2,3,4,1)
move_and_capture(2,3,4,2)
move_and_capture(2,3,4,3)
move_and_capture(2,3,4,4)
moves(2,3)


