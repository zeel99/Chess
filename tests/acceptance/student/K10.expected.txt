  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
->setup_chess(P,1,1)
  # of chess pieces on board: 1
  Game being Setup...
  P...
  ....
  ....
  ....
->setup_chess(Q,2,3)
  # of chess pieces on board: 2
  Game being Setup...
  P...
  ..Q.
  ....
  ....
->setup_chess(P,3,1)
  # of chess pieces on board: 3
  Game being Setup...
  P...
  ..Q.
  P...
  ....
->setup_chess(P,4,2)
  # of chess pieces on board: 4
  Game being Setup...
  P...
  ..Q.
  P...
  .P..
->setup_chess(P,4,4)
  # of chess pieces on board: 5
  Game being Setup...
  P...
  ..Q.
  P...
  .P.P
->start_game
  # of chess pieces on board: 5
  Game In Progress...
  P...
  ..Q.
  P...
  .P.P
->move_and_capture(2,3,1,1)
  # of chess pieces on board: 5
  Error: Invalid move of Q from (2, 3) to (1, 1)
  P...
  ..Q.
  P...
  .P.P
->move_and_capture(2,3,1,2)
  # of chess pieces on board: 5
  Error: Slot @ (1, 2) not occupied
  P...
  ..Q.
  P...
  .P.P
->move_and_capture(2,3,1,3)
  # of chess pieces on board: 5
  Error: Slot @ (1, 3) not occupied
  P...
  ..Q.
  P...
  .P.P
->move_and_capture(2,3,1,4)
  # of chess pieces on board: 5
  Error: Slot @ (1, 4) not occupied
  P...
  ..Q.
  P...
  .P.P
->move_and_capture(2,3,2,1)
  # of chess pieces on board: 5
  Error: Slot @ (2, 1) not occupied
  P...
  ..Q.
  P...
  .P.P
->move_and_capture(2,3,2,2)
  # of chess pieces on board: 5
  Error: Slot @ (2, 2) not occupied
  P...
  ..Q.
  P...
  .P.P
->move_and_capture(2,3,2,3)
  # of chess pieces on board: 5
  Error: Invalid move of Q from (2, 3) to (2, 3)
  P...
  ..Q.
  P...
  .P.P
->move_and_capture(2,3,2,4)
  # of chess pieces on board: 5
  Error: Slot @ (2, 4) not occupied
  P...
  ..Q.
  P...
  .P.P
->move_and_capture(2,3,3,1)
  # of chess pieces on board: 5
  Error: Invalid move of Q from (2, 3) to (3, 1)
  P...
  ..Q.
  P...
  .P.P
->move_and_capture(2,3,3,2)
  # of chess pieces on board: 5
  Error: Slot @ (3, 2) not occupied
  P...
  ..Q.
  P...
  .P.P
->move_and_capture(2,3,3,3)
  # of chess pieces on board: 5
  Error: Slot @ (3, 3) not occupied
  P...
  ..Q.
  P...
  .P.P
->move_and_capture(2,3,3,4)
  # of chess pieces on board: 5
  Error: Slot @ (3, 4) not occupied
  P...
  ..Q.
  P...
  .P.P
->move_and_capture(2,3,4,1)
  # of chess pieces on board: 5
  Error: Slot @ (4, 1) not occupied
  P...
  ..Q.
  P...
  .P.P
->move_and_capture(2,3,4,2)
  # of chess pieces on board: 5
  Error: Invalid move of Q from (2, 3) to (4, 2)
  P...
  ..Q.
  P...
  .P.P
->move_and_capture(2,3,4,3)
  # of chess pieces on board: 5
  Error: Slot @ (4, 3) not occupied
  P...
  ..Q.
  P...
  .P.P
->move_and_capture(2,3,4,4)
  # of chess pieces on board: 5
  Error: Invalid move of Q from (2, 3) to (4, 4)
  P...
  ..Q.
  P...
  .P.P
->moves(2,3)
  # of chess pieces on board: 5
  Game In Progress...
  .+++
  ++Q+
  .+++
  +.+.
